### DISCUSSION TITLE
legacy `some` vs proposes `any`

### DISCUSSION THREAD

**Discussion by crusso** [caffeinelabs/motoko-core#60] [Ideas]
`all`/`some` vs `all`/`any` has the advantage that `some` is fairly standard in logic  and less likely to be confused with `all` than `any`. 

Another option might be `exists`, or `forAll`, `forSome` but might be worth checking if there were any guidelines...

**Comment by rvanasa** [comment]
Python, Java/Kotlin, C# (LINQ), Haskell, Rust, C++, Julia, and Ruby use the names `any` and `all`, so this seems to be the agreement among most commonly-used languages at this point.

JavaScript and Common Lisp use `some` and `every` as a different pair of names. 

We could keep the original names, but there would be some value in choosing either (`any`, `all`) or (`some`, `every`).

Another reason for changing `some` to `any` is that this could potentially be confusing with `Option.some()` having a different purpose.

**Reply by crusso** [reply]
Ok, let's go with crowd then. I didn't realize any was so common..
